apps/sms/views/shared/js/settings.js

```diff
@@ -67,9 +70,37 @@ var Settings = {
       }
     }
 
+    //create lock for SEND_READ_REPORT_KEY
+    keyHandlerSet[this.SEND_READ_REPORT_KEY] = this.toSendReadReport.bind(this);
+
     for (var key in keyHandlerSet) {
       setHandlerMap(key);
     }
+
+    //add Observer for SEND_READ_REPORT_KEY
+    settings.addObserver(this.SEND_READ_REPORT_KEY, function (event) {
+      Settings.sendReadReportStatus = event.settingValue;
+      console.log(Settings.sendReadReportStatus);
+    });
```

Comment:
you can add hte observer in "initSendReadReport" like the other "init*" functions